src/core/index.js

```diff
@@ -79,42 +95,23 @@ let Core = function( opts ){
       current: [],
       queue: []
     },
+    selectionType: options.selectionType,
     hasCompoundNodes: false
   };
 
-  this.createEmitter();
-
-  // set selection type
-  let selType = options.selectionType;
-  if( selType === undefined || (selType !== 'additive' && selType !== 'single') ){
-    // then set default
+  return _p;
+};
 
-    _p.selectionType = 'single';
-  } else {
-    _p.selectionType = selType;
-  }
+let coreInit = function( cy, opts ){
+  let options = createInitOptions( opts );
+  let _p = cy._private = createCore( options );
 
-  // init zoom bounds
-  if( is.number( options.minZoom ) && is.number( options.maxZoom ) && options.minZoom < options.maxZoom ){
-    _p.minZoom = options.minZoom;
-    _p.maxZoom = options.maxZoom;
-  } else if( is.number( options.minZoom ) && options.maxZoom === undefined ){
-    _p.minZoom = options.minZoom;
-  } else if( is.number( options.maxZoom ) && options.minZoom === undefined ){
-    _p.maxZoom = options.maxZoom;
-  }
+  _p.elements = new Collection( cy );
```

Comment:
Not sure if I should be creating a new Collection every time coreInit (i.e mount) is called, or if this should only be called once during the Core constructor.